packages/cli/src/ui/App.tsx

```diff
@@ -455,26 +531,20 @@ const App = ({ config, settings, startupWarnings = [], version }: AppProps) => {
       setConstrainHeight(true);
     }
 
-    if (key.ctrl && input === 'o') {
-      setShowErrorDetails((prev) => !prev);
-    } else if (key.ctrl && input === 't') {
-      const newValue = !showToolDescriptions;
-      setShowToolDescriptions(newValue);
+    const context = { enteringConstrainHeightMode };
 
-      const mcpServers = config.getMcpServers();
-      if (Object.keys(mcpServers || {}).length > 0) {
-        handleSlashCommand(newValue ? '/mcp desc' : '/mcp nodesc');
-      }
-    } else if (key.ctrl && (input === 'c' || input === 'C')) {
-      handleExit(ctrlCPressedOnce, setCtrlCPressedOnce, ctrlCTimerRef);
-    } else if (key.ctrl && (input === 'd' || input === 'D')) {
-      if (buffer.text.length > 0) {
-        // Do nothing if there is text in the input.
-        return;
+    const handler = keystrokeHandlers.find((h) => {
```

Comment:
this logic to execute the keystroke handlers seems independent on app. I would move it to keystrokeHandler.ts (also please rename that file to align with naming schemes for non tsx files in this project.
Would then be nice to add a basic unittest for it that verifies cases such as if key.shift  and key.ctrl are handled correctly. seems like right now key.shift is ignored.